### ISSUE TITLE
moretags-ish for messages

### ISSUE THREAD

**Issue by travisjeffery** [gogo/protobuf#397]
What do you think about moretags or something like it for messages. The use case if you have a message like:

``` proto
message Person {
  string first_name = 1;
  string last_name = 2;
  uint32 age = 3;
}
```

And you want all the fields to have the same tag, for instance if you're using sqlx you'd want:

``` proto
message Person {
  string first_name = 1 [(gogoproto.moretags) = "db:\"first_name\""];
  string last_name = 2 [(gogoproto.moretags) = "db:\"last_name\""];
  uint32 age = 3 [(gogoproto.moretags) = "db:\"age\""];
}
```

But it verbose and ugly to write that out for each field, it'd be nice if you could do the following which would be equivalent:

``` proto
message Person {
 [(gogoproto.moretags) = "db"];

  string first_name = 1;
  string last_name = 2;
  uint32 age = 3;
}
```

It could take a parameter for the tag name style e.g. "underscore" vs. "camelcase".

I can write this and I will anyway but was just asking for your input and whether you think this would fit into gogo proto.

**Comment by awalterschulze** [comment]
Sounds a lot like this idea
https://github.com/gogo/protobuf/pull/368

Personally this is where I prefer a vanity binary to fill in the moretags for me, but I am not totally opposed to finding a solution here, I just think its hard.

If I remember correctly `omitempty` is where the previous proposal fell apart.
I would like to only add one option, without any magic that covers a big chunk of usecases.

**Comment by travisjeffery** [comment]
Yeah looks like #368 might cover it so you'd do `option (gogoproto.copyjsontagto_all) = "db";` in this case. Does that look like it's gonna be merged?

The vanity binary is what I do now, but with so many people writing vanity binaries to do the same thing would be nice to have one thing everyone could use.

**Comment by awalterschulze** [comment]
I agree that it would be nice to solve with something everyone can use, but we are stuck on the omitempty issue, which might require two solutions and then it also feels wrong.

Do you have any thoughts on how to solve omitempty?

**Comment by justjoeyuk** [comment]
I'd recommend having the `option (gogoproto.copyjsontagto_all) = "db";` but then in the individual fields you could have:

`string my_field [(gogo.omitempty="bson,json")]` or something?

**Comment by awalterschulze** [comment]
Then it could just be `option (gogoproto.copyfieldname_all) = "db";`, since json tags is something that golang/protobuf wants to drop in future.

And maybe we could extend fields with a list instead of a string that needs to be parsed:
`[(gogo.omitempty=["bson","json"])]`
But I don't know the real syntax for that.

Hmmm

**Comment by justjoeyuk** [comment]
Yeah that would work. I'd just be afraid of the inconsistencies unless everything would be changed to a list rather than a string?

I must admit that I prefer the list format.

**Comment by awalterschulze** [comment]
I also like to avoid inconsistencies, if possible.
Which inconsistencies are you referring to?

**Comment by justjoeyuk** [comment]
There might not be any. I just mean if there is anything at the moment that is a comma-separated string - it should probably be switched to follow this array format.

**Comment by awalterschulze** [comment]
Cool.  A bit hard to change things like that in gogoprotobuf, but I also hope there aren't any yet.

Any other opinions on this suggestion?
